### PR TITLE
BACKLOG-22323: Implement a renderValue property for richtext and also fix QA-15088

### PR DESCRIPTION
**Pull request by jayblanc**
## JIRA

https://jira.jahia.org/browse/BACKLOG-22323
https://jira.jahia.org/browse/QA-15088

## Description

The goal is to add a GraphQL Property Extension that will render richtext that includes internal URLs, Vanity URLs, Macro, in order to provide a pre rendered version of the richtext content in GraphQL.
It also fix a bug that breaks the Node rendering extension. 
It is mostly based on calling a configured RenderEngine allowing filtering of embedded content.

## Tests

The following are included in this PR
- [X] Unit Tests (Most changes _should_ have unit tests)
- [ ] Integration Tests

## Checklist

I have considered the following implications of my change: 

- [X] Security (in particular for changes to authentication, authorization, data fetching, ...)
- [X] Performance
- [X] Migration
- [X] Code maintainability

## Documentation

- [ ] Inline documentation
- [ ] Internal Documentation (wiki)
- [ ] User-facing Documentation

### CODE REVIEW

File: graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/relay/PaginationHelper.java [hunk 1]
```diff
@@ -152,10 +162,27 @@ private static <T> List<T> collectItems(Iterator<T> it, Arguments arguments, Cur
             if (arguments.before != null && cursorSupport.getCursor(value).equals(arguments.before)) {
                 break;
             }
+
+            //Adding a limit of 1000 to avoid OOM
```
Comment (sergehuber) [inline graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/relay/PaginationHelper.java:166]: I think this comment is not up to date ?

File: graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/render/RenderPropertyExtensions.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+public class RenderPropertyExtensions {
+
+    private GqlJcrProperty property;
+
+    public RenderPropertyExtensions(GqlJcrProperty property) {
+        this.property = property;
+    }
+
+    @GraphQLField
+    @GraphQLDescription("Gets the rendered content values of that node")
```
Comment (sergehuber) [inline graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/render/RenderPropertyExtensions.java:66]: shouldn't it be "of that property" ?

File: graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/render/RenderPropertyExtensions.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+                result.add(renderValue("html", property.getLanguage(), value, environment.getGraphQlContext()));
+            }
+            return result;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @GraphQLField
+    @GraphQLDescription("Gets the rendered content value of that node")
```
Comment (sergehuber) [inline graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/render/RenderPropertyExtensions.java:85]: Same here : property

File: graphql-dxm-provider/src/main/resources/META-INF/configurations/org.jahia.modules.graphql.provider-default.cfg [hunk 4]
```diff
@@ -33,3 +33,4 @@
 # List of allowed origin for CORS access :
 #
 # http.cors.allow-origin=http://mysite1.com, http://mysite2.com
+graphql.fields.node.limit = 5000
```
Comment (sergehuber) [inline graphql-dxm-provider/src/main/resources/META-INF/configurations/org.jahia.modules.graphql.provider-default.cfg:36]: Maybe add a comment as to what this setting does ?

**Review by sergehuber — CHANGES_REQUESTED** [review]
Hi Jérome, thanks for the PR. 
I found some minor issues but apart from that nice work !

File: graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/relay/PaginationHelper.java (hunk 1 above)
Comment (jayblanc) [reply graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/relay/PaginationHelper.java:166]: This is from a rebase, not originally in my PR...

File: graphql-dxm-provider/src/main/resources/META-INF/configurations/org.jahia.modules.graphql.provider-default.cfg (hunk 4 above)
Comment (jayblanc) [reply graphql-dxm-provider/src/main/resources/META-INF/configurations/org.jahia.modules.graphql.provider-default.cfg:36]: fixed

File: graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/render/RenderPropertyExtensions.java (hunk 3 above)
Comment (jayblanc) [reply graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/render/RenderPropertyExtensions.java:85]: fixed

**Review by sergehuber — APPROVED** [review]
Thanks for the changes, looks good now, I'm approving the PR

**Review by sergehuber — APPROVED** [review]

**Pull request merged**